src/bp/sdk/worker-before.ts

```diff
@@ -0,0 +1,38 @@
+import { EventEmitter } from 'events'
+import path from 'path'
+import { workerData } from 'worker_threads'
+
+import { Distro } from '../common/getos'
+
+process.BOTPRESS_EVENTS = new EventEmitter()
+process.BOTPRESS_EVENTS.setMaxListeners(1000)
+global.BOTPRESS_CORE_EVENT = (event, args) => process.BOTPRESS_EVENTS.emit(event, args)
+
+const processData = workerData?.processData
+if (processData) {
+  Object.assign(process, processData)
+  process.distro = new Distro(JSON.parse(processData.distro))
+}
+
+if (workerData?.processEnv) {
+  Object.assign(process.env, workerData.processEnv)
+  process.core_env = process.env as BotpressEnvironmentVariables
+}
+
+if (!process.core_env) {
+  if (!!process.env.pkg) {
+    process.pkg = process.env.pkg
```

Comment:
why that check ? you use both var in your ternary

oh. I imagine you didn't write this, simply moved it in dedicated file ?

Reply:
I actually copied this file from a branch Sly wrote few weeks ago. I havent cleaned it, but I should.